src/Forms/PrefabFormFactory.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/*
+Gibbon, Flexible & Open School System
+Copyright (C) 2010, Ross Parker
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+namespace Gibbon\Forms;
+
+/**
+ * PrefabFormFactory
+ *
+ * For building pre-made forms all at once.
+ *
+ * @version v14
+ * @since   v14
+ */
+class PrefabFormFactory
+{
+    public static function createDeleteForm($action, $confirmation = false)
+    {
+        $form = Form::create('deleteRecord', $action);
+        $form->addHiddenValue('address', $_GET['q']);
```

Comment:
Why putting the value in address too, it'll be injected as `q` in line 38 anyway?

Reply:
Hi Marc! A good question. Many of the deleteProcess pages rely on an address value available in POST, the added line here prevents breaking the existing behaviour in the code.

Reply:
Cool, just wanted to make sure this was intentional and not just cargo cult.  I'm JUST starting into the codebase and PHP is not my native/best language... sorry for any questions.

Reply:
Questions are most welcome! This class is part of our ongoing effort to modernize the core, focusing in v15 on moving the html forms over to a set of OO classes. Feel free to give a shout as you delve into the codebase, also be sure to check out the support forum: ask.gibbonedu.org

Reply:
Hi @IDisposable thanks for stopping in to comment. I'm not sure we've met before (I may be mistaken), so welcome to the Gibbon community : )